SalesforceCommerceCloud/pwa-kit
# Add guards to result sent to Einstein

# Description




The PR adds guards to recent Einstein changes made in https://github.com/SalesforceCommerceCloud/pwa-kit/pull/714

On a generated project with preset `retail-react-app-demo`, navigating from Home to PLP was showing two errors:

Error 1:
```
einstein.js:152 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'hits')
    at einstein.js:152:40
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (asyncToGenerator.js:3:1)
    at _next (asyncToGenerator.js:25:1)
    at asyncToGenerator.js:32:1
    at new Promise (<anonymous>)
    at asyncToGenerator.js:21:1
    at EinsteinAPI.sendViewCategory (einstein.js:148:59)
    at useEinstein.js:29:33
    at Generator.next (<anonymous>)
```

Error 2:
```
POST http://localhost:3000/mobify/proxy/einstein/v3/activities/aaij-MobileFirst/viewCategory 400 (Bad Request)

{"error":{"code":"ValidationError","msg":"There was a problem validating the request. See `errors`.","errors":[{"msg":"Must provide a `products` array.","param":"products","location":"body"}]}}
```

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Added result guards to Einstein and to PLP sendViewCategory Einstein

# How to Test-Drive This PR

Steps to reproduce the issue:
1. Generate a project with `retail-react-app-demo` preset.

```
npx pwa-kit-create-app@2.3.0-alpha.0 --preset retail-react-app-demo
```

2. Navigate from Home page to PLP page.
3. Observe the errors in the browser console. 

Verify the errors are not shown running the same steps on a generated project from this branch.
```
GENERATOR_PRESET=retail-react-app-demo node packages/pwa-kit-create-app/scripts/create-mobify-app-dev.js --outputDir /tmp/generated-project
```

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
